src/app.vue

```diff
@@ -0,0 +1,143 @@
+							</a>
+						</li>
+						<li ng-click="setSortOrder($event, 'start')" ng-class="{active: settingsmodel.getById('various').sortOrder == 'start'}">
+							<a>
+								<span class="icon icon-calendar" />
+								<span class="label">{{ t('tasks', 'Start date') }}</span>
+								<span class="icon sort-indicator" ng-class="{'icon-sort-up': settingsmodel.getById('various').sortDirection, 'icon-sort-down': !settingsmodel.getById('various').sortDirection}" />
+							</a>
+						</li>
+						<li ng-click="setSortOrder($event, 'priority')" ng-class="{active: settingsmodel.getById('various').sortOrder == 'priority'}">
+							<a>
+								<span class="icon icon-task-star" />
+								<span class="label">{{ t('tasks', 'Priority') }}</span>
+								<span class="icon sort-indicator" ng-class="{'icon-sort-up': settingsmodel.getById('various').sortDirection, 'icon-sort-down': !settingsmodel.getById('various').sortDirection}" />
+							</a>
+						</li>
+						<li ng-click="setSortOrder($event, 'alphabetically')" ng-class="{active: settingsmodel.getById('various').sortOrder == 'alphabetically'}">
+							<a>
+								<span class="icon icon-alphabetically" />
+								<span class="label">{{ t('tasks', 'Alphabetically') }}</span>
+								<span class="icon sort-indicator" ng-class="{'icon-sort-up': settingsmodel.getById('various').sortDirection, 'icon-sort-down': !settingsmodel.getById('various').sortDirection}" />
+							</a>
+						</li>
+						<!-- <li ng-click="setSortOrder($event, 'manual')" ng-class="{active: settingsmodel.getById('various').sortOrder == 'manual'}">
+							<a>
+								<span class="icon icon-manual" />
+								<span class="label">{{ t('tasks', 'Manually') }</span>
+								<span class="icon sort-indicator" ng-class="{'icon-sort-up': settingsmodel.getById('various').sortDirection, 'icon-sort-down': !settingsmodel.getById('various').sortDirection}" />
+							</a>
+						</li> -->
+					</ul>
+				</div>
+				<div class="task-list">
+					<router-view />
+					<div id="searchresults" />
+					<div class="task-item template">
+						<div class="task-body">
+							<div class="percentdone" />
+							<a class="task-checkbox" name="toggleCompleted" ng-click="toggleCompleted()">
+								<span class="icon task-checkbox" />
+							</a>
+							<a class="icon task-separator" />
+							<a class="task-star" ng-click="toggleStarred(task.id)">
+								<span class="icon task-star faded" />
+							</a>
+							<!-- <a class="duedate" ng-class="{overdue: TasksModel.overdue(task.due)}">{{ task.due | dateTaskList }}</a> -->
+							<div class="title-wrapper">
+								<span class="title" />
+								<span class="icon task-attachment" />
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<div id="app-sidebar">
+			<theDetails />
+		</div>
+	</div>
+</template>
+
+<script>
+import TheList from './components/TheList'
```

Comment:
We already have a component for the Navigation :)
https://github.com/nextcloud/vue-components/tree/init

Reply:
Thanks for the hint. Do you use them in the contacts rewrite already? I can't find how you imported them there?

Reply:
Yes, we copied them directly because we did not published them to npm yet. But this should be done next week :)
Sorry about the late comment, i thought I already give you this information :(

You can also see that we imported them into the vueexample app (see package.json). If you use the same components from this repo the transition to import should be very straightforward anyway ;)

Reply:
I guess a
`
"vue-components": "git+https://github.com/nextcloud/vue-components.git#init",
`
should do for now. I will have a look how they work.

Reply:
A problem with the popover component will be, that I cannot use a custom item there. The delete item is currently showing an element asking if you really want to delete: https://github.com/nextcloud/tasks/blob/master/js/app/directives/confirmation.js. That will not work with the standard popover menu.

Reply:
You should not use any custom item according to our standard though :)
That's why we created standards :see_no_evil: 

The timeout is not necessary if you already have an undo option.

Reply:
And if you want a confirmation, just change the object item text and ask for a second click action, still within the same object :)

Reply:
Arf, sorry, I thought you had an undo function! My bad! :)

Well, the components use a dedicated prop which is an object.
After clicking the delete action, you can just trigger a change that replace the object entry related to the delete action by a confirm one that will then trigger the final deletion :)

You can even have the countdown if you want, when using a computed variable, it should be feasible!

Exampel of dynamic checks to build the menu: https://github.com/nextcloud/contacts/blob/vue/src/components/Settings/SettingsAddressbook.vue#L129
Or dynamic text: 
https://github.com/nextcloud/contacts/blob/vue/src/components/Settings/SettingsAddressbook.vue#L102

Reply:
Rough example
``` js
			menu = [
				{
					icon: 'icon-public',
					text: this.confirm
						? t('tasks', 'Are you sure')
						: t('tasks', 'Delete'),
					action: this.confirm
						? this.delete
						: this.toggleConfirm
				}
			]
			return menu
 ```

Reply:
Thanks a lot, I will have a look. 😃